### ISSUE TITLE
Missing links

### ISSUE THREAD

**Issue by afabri** [CGAL/cgal#7839]
## Issue Details

I am wondering what we can do on the missing links for  `CGAL::is_closed(tm)` [here](https://doc.cgal.org/latest/Polygon_mesh_processing/group__PMP__orientation__grp.html#gabdb85ef9ccc1a4da632f1a697c6d5163).

Is this because we are in the subnamespace `CGAL::Polygon_mesh_processing`?   Is the `tm` the problem? 

It is the same for doxygen master  [here](https://cgal.geometryfactory.com/CGAL/Manual_doxygen_test/CGAL-6.0-Ic-96/master/Polygon_mesh_processing/group__PMP__orientation__grp.html#gabdb85ef9ccc1a4da632f1a697c6d5163)

@albert-github do you have any idea or suggestion?

**Comment by albert-github** [comment]
Yes, the `tm` is the problem as this is the, dummy / placeholder, name of the argument. One should either, in this case, use no arguments at all or use the type. I did a small test with
```
 * @pre `CGAL::is_closed(tm)`
 * @pre `CGAL::is_closed`
 * @pre `CGAL::is_closed()`
 * @pre `CGAL::is_closed(const TriangleMesh&)`
```
in include/CGAL/Polygon_mesh_processing/orientation.h and the result was:
![image](https://github.com/CGAL/cgal/assets/5223533/2367681f-53fc-42c5-ab9b-0a3be332a383)

where the last 3 `is_closed` are hyperlinked.

**Comment by afabri** [comment]
As we want to express a precondition the `tm`  is important here.   So we probaby have to use a doxygen `link` command?

**Comment by albert-github** [comment]
Yes like:
```
 * @pre `CGAL::is_closed(tm)`
 * @pre \link CGAL::is_closed `CGAL::is_closed(tm)` \endlink
```
result:
![image](https://github.com/CGAL/cgal/assets/5223533/f14b2eaa-3ccc-4c51-a3d9-915db933a378)

or use a descriptive text, as the usage if `tm` (is a bit contradicting the normal usage of the links by using the "dummy / placeholder name of the argument". Furthermore the `tm` is for a C++ programmer a bit confusing as it could also refer to a `tm` time type of struct see e.g. https://cplusplus.com/reference/ctime/tm/)

**Comment by afabri** [comment]
I would go for the `\link`.   I don't see a problem with the `tm`  of `<ctime>`.    We are in the scope of the explanation of the function and we have introduced `tm`  as a `\param`   in this scope.   And again, it seems normal that a precondition  is an expression that concerns the input of the function.  Thanks for having investigated.   And you can confirm that this is not a doxygen issue that might get fixed in the future?

**Comment by albert-github** [comment]
Where did you see the link. Probably the reason is that in the example only the part `is_closed` is hyperlinked in the precondition the full word is attempted to link

**Comment by sloriot** [comment]
See https://doc.cgal.org/latest/Polygon_mesh_processing/Polygon_mesh_processing_2orient_polygon_soup_example_8cpp-example.html

**Comment by albert-github** [comment]
I assume you mean here the links of the function calls in the main function.
The links in the comment and the links in the code (example) are handled differently. In the comment what is more "natural speech"  the separators are most of the time characters like spaces, commas etc. whilst in the code brackets etc. also have an effect.

**Comment by MaelRL** [comment]
> I assume you mean here the links of the function calls in the main function. The links in the comment and the links in the code (example) are handled differently. In the comment what is more "natural speech" the separators are most of the time characters like spaces, commas etc. whilst in the code brackets etc. also have an effect.

But whether we are in \pre or in generic doc, we are always surrounded by backticks in these missing link cases, so it's not really natural speech but clearly code.

Anyway I started adding in https://github.com/CGAL/cgal/pull/9074, but there is really a ton... 

Also, there is a difference in rendering (no underline with \link), e.g. :

```cpp
\sa \link compare_z_grp `CGAL::compare_z()` \endlink
\sa `CGAL::compare_theta_z()`
```
gives

<img width="307" height="67" alt="Image" src="https://github.com/user-attachments/assets/06e59b88-e968-4f31-857c-3ec28f41aea6" />

**Comment by albert-github** [comment]
As far as I know, by head, the fact that the `` `...` `` part as of the link and the presence of the underline has to do with the order of  and the used CSS items, it indeed looks a bit unexpected.
I'll have to dive into it (but I can have a look first in a couple of weeks time)

**Comment by MaelRL** [comment]
Thank you @albert-github, much appreciated!

**Comment by albert-github** [comment]
Is this issue still relevant (as #9074 has been integrated) and in CGAL-6.2-Ic-46 on the page Polygon_mesh_processing/group__PMP__orientation__grp.html#gabdb85ef9ccc1a4da632f1a697c6d5163 I see both links present.
Or are there still other places to be considered? If so where?

**Comment by MaelRL** [comment]
Yes it's still relevant (also the underline thingy), I fixed manually in PMP & a few other packages, but some I have not done.

Thanks!
